sourced/ml/uast_struct_to_bag.py

```diff
@@ -0,0 +1,164 @@
+from collections import defaultdict
+from random import random, choice
+
+from sourced.ml.uast_ids_to_bag import FakeVocabulary
+
+
+class UASTStructure2BagBase:
+    def uast_to_bag(self, uast):
+        raise NotImplemented
+
+
+class UAST2Seq2Bag(UASTStructure2BagBase):
+    """
+    DFS traversal + preserve order of node children.
```

Comment:
Why DFS? As I remember we randomly walk.

Reply:
it's another approach that traverses tree to preserve the order of nodes as it should be in code.